src/transformers/models/mistral/modeling_mistral.py

```diff
@@ -346,13 +344,14 @@ def forward(
         use_cache: bool = False,
         **kwargs,
     ):
-        if "padding_mask" in kwargs:
-            warnings.warn(
-                "Passing `padding_mask` is deprecated and will be removed in v4.37. Please make sure use `attention_mask` instead.`"
+        if isinstance(past_key_value, StaticCache):
+            raise ValueError(
+                "`static` cache implementation is not compatible with `attn_implementation==flash_attention_2` "
+                "make sure to use `sdpa` in the mean time, and open an issue at https://github.com/huggingface/transformers"
```

Comment:
since we remove the padding mask here, let's remove the kwargs, same for all attention class.